Wacom Support, absolute x,y pressure; relative scroll; forward/back buttons; platform address width case for event timestamp

olewhalehunter opened the pull request:
Wacom Support, absolute x,y pressure; relative scroll; forward/back buttons; platform address width case for event timestamp

library will not work on 32 bit systems without unsigned-long-integer case fix included in this commit

tested on Wacom CTE-440 and Lenovo Black Silk keyboard

olewhalehunter committed:
wacom support for absolute, relative, and forward/back buttons, 32/64bit unix time case

olewhalehunter committed:
wacom support for absolute, relative, and forward/back buttons, 32/64bit unix time case

jtgans commented on evdev.lisp:
```diff
@@ -154,25 +154,47 @@ used outside of this package.")
+)
+
+
+(cond ((equal (machine-type) "X86") 
```
This concerns me a bit -- with this change, it'll only work on x86-based machines. ARM boxen are effectively out.

jtgans commented on evdev.lisp:
```diff
@@ -154,25 +154,47 @@ used outside of this package.")
   (:documentation "This is a verbaticm copy of the input_event struct defined in
 linux/include/uapi/linux/input.h. This is used to read in each event straight
-from raw evdev data."))
+from raw evdev data. Unix time values bit widths differ on 32/64bit systems."))
```
We'll want to note here that the difference is based upon the return value of machine-type.

jtgans reviewed (changes requested):
Adjust the comment and I'll happily take this change. =o)

olewhalehunter committed:
input-event-struct time bit width documentation

inline unix time bit width documentation on input-event-struct adjusted to clarify SBCL (machine-type) contingency

olewhalehunter commented:
Documentation adjusted in input-event-struct per requested changes.
